### DISCUSSION TITLE
Linking to Lia Courses without the preview

### DISCUSSION THREAD

**Discussion by tilman-schieber** [LiaScript/LiaScript#207] [Q&A]
I am using one md-File that is just a `<lia-preview>` of all the chapters in my course. 
Now I want to add a backlink on each course site, so students can navigate back to the overview page.
This should just be a normal link, not a preview card. 

But....
-  if I use a relative link to the md-File it is not interpreted as lia.
- if I use a full-length link to `liascript.github.io/course=fullpathtogithub` it works but will probably break in the future or if the course gets forked

Is there a way to do it with the current features?
Or what about a lia-link in addition to lia-preview?

**Comment by abotzki** [comment]
This is an interesting comment which we are also discussing at the moment internally.

We have colleagues who prefer to have individual chapters of a course in separate markdown files. 

The whole course information, exercises, small data sets are in the same github repo. 
As a side note: Via github search, you can of course search through all of the chapters. Obviously, by splitting the content in different markdown files, overarching content searching in a rendered course is not possible.

If you have course overview page using Liascript, once you go to the chapters (i.e. different full-length link like above) you loose the context to the overview page.

So, I would also be interested in hearing comments from others on this situation @tilman-schieber describes. Thank you!

**Reply by cagix** [reply]
> We have colleagues who prefer to have individual chapters of a course in separate markdown files.
> The whole course information, exercises, small data sets are in the same github repo.
> So, I would also be interested in hearing comments from others on this situation

That's a good description of my workflow! I have an overview page in my repo that contains the syllabus and a roadmap, and then individual Markdown files for each chapter. These individual chapters are linked from the roadmap via a local Markdown links. There are also exercises, code snippets and other data. 

At the end, I would like to share the overview page as a Lia Course with the students, basically as a start page for the course.

**Comment by tilman-schieber** [comment]
One additional aspect:

Having an overview course with lia-preview cards is a really nice way to show the chapters of a course.
The different logos and comments are nicely rendered and allow easy navigation (at least in one direction).

see [the overview here](https://liascript.github.io/course/?https://raw.githubusercontent.com/tilman-schieber//OKInf/main/0_Inhalt.md) for an example

For larger projects, I find it extremely clumsy to have all course content in on markdown file. One big file complicates version management and gradually publishing course content for students.
lia-preview is uniquely suited for displaying chapters, and I can hardly imagine a better use case for it. It is after all quite rare that you link to unrelated courses.

**Comment by andre-dietrich** [comment]
We all agreed that the URL liascript.github.io will remain, since no one wants to pay ;-) ... To solve your issue, you could probably use something like a custom macro for this:


``` markdown
<!--
@lia_link
<script modify="false" run-once="true">
`LIASCRIPT: [@0](${window.location.origin + window.location.pathname}?@1)`
</script>
@end
-->

# Back-link

@[lia_link(relative)](dir/french.md)

@[lia_link(absolute)](https://example.com/README.md)
```

In this case the course can also be delivered from an exported web-project or scrom... It it will use the base URL as the origin and attach the markdown, when it is relative or absolute ...

However ... Constructing larger projects from smaller files was also an issue for us, that we could not solve yet. One Idea would be to add such a functionality into the exporter, which we currently use to generate overview sites for the course, take look at here:

https://tubaf-ifi-liascript.github.io/

Such sites can be automatically generated with the exporter ...

We did not att something like summary.md from GitBook, since it breaks or makes the parsing process more complicated. One document is cached and updated, where we use the URL or URI as the unique identifier for indexeddb ... handling multiple files will break this ...

How we internaly store content is described in the following section:

https://liascript.github.io/course/?https://raw.githubusercontent.com/liaScript/docs/master/README.md#version

you can use the `@version` to mark changes, thus, adding new chapters sequentially will not break LiaScript, you can update this without a problem. Only remixing slides does require a major - version change ...

Not sure, if this solves your issue???

However, your courses look great ... If you want to, we would be happy to mark them on the LiaScript-map:

https://liascript.github.io/world-map/

**Reply by tilman-schieber** [reply]
Thanks for the macro. Looks like a perfect solution. 

> However, your courses look great ... If you want to, we would be happy to mark them on the LiaScript-map:

Thanks, they are a work in progress. If you don't mind adding unfinished stuff, add them anytime!
